clustered/integration-test/src/test/java/org/ehcache/clustered/replication/OversizedCacheOpsPassiveTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.replication;
+
+import org.ehcache.Cache;
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.ClusteredTests;
+import org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.terracotta.testing.rules.Cluster;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+
+import static org.terracotta.testing.rules.BasicExternalClusterBuilder.newCluster;
+
+/**
+ * Test the effect of cache eviction during passive sync.
+ */
+public class OversizedCacheOpsPassiveTest extends ClusteredTests {
+  private static final int MAX_PUTS = 3000;
+  private static final int MAX_SWITCH_OVER = 3;
+  private static final int PER_ELEMENT_SIZE = 256 * 1024;
+  private static final int CACHE_SIZE_IN_MB = 2;
+  private static final String LARGE_VALUE = buildLargeString();
+
+  private static final String RESOURCE_CONFIG =
+      "<config xmlns:ohr='http://www.terracotta.org/config/offheap-resource'>"
+      + "<ohr:offheap-resources>"
+      + "<ohr:resource name=\"primary-server-resource\" unit=\"MB\">2</ohr:resource>"
+      + "</ohr:offheap-resources>" +
+      "</config>\n";
+
+  @ClassRule
+  public static Cluster CLUSTER =
+      newCluster(2).in(Paths.get("build", "cluster").toFile())
+        .withSystemProperty("ehcache.sync.data.gets.threshold", "2")
+        .withServiceFragment(RESOURCE_CONFIG)
+        .build();
+
+  @BeforeClass
+  public static void waitForServers() throws Exception {
```

Comment:
@chrisdennis I had to do a waitForServers for this test case because otherwise the terminateActive that happens immediately in the test is taking platform through different paths which is not the purpose of this test, On slow build machines, this is making the core hit different windows.

Reply:
Okay - makes sense.